foodsaving/groups/models.py

```diff
@@ -37,6 +38,12 @@ class Group(BaseModel, LocationModel, ConversationMixin):
         on_delete=models.SET_NULL
     )
 
+    def approved_member_count(self):
+        return self.members_with_all_roles([GROUP_APPROVED_MEMBER]).count()
+
+    def members_with_all_roles(self, roles):
+        return self.members.filter(groupmembership__roles__contains=roles)
```

Comment:
I removed the original `members_with_*`  method, in favour of a more reusable approach, although maybe it's not clear yet how to do that and if it's better.